pkg/controller/controller.go

```diff
@@ -112,46 +89,65 @@ func NewController(shareRelist time.Duration) (*Controller, error) {
 		shareRelist)
 
 	c := &Controller{
+		kubeClient:     kubeClient,
 		kubeRestConfig: kubeRestConfig,
-		cfgMapWorkqueue: workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(),
-			"shared-resource-configmap-changes"),
-		secretWorkqueue: workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(),
-			"shared-resource-secret-changes"),
 		shareWorkqueue: workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(),
 			"shared-resource-share-changes"),
 		informerFactory:      informerFactory,
 		shareInformerFactory: shareInformerFactory,
-		cfgMapInformer:       informerFactory.Core().V1().ConfigMaps().Informer(),
-		secInformer:          informerFactory.Core().V1().Secrets().Informer(),
 		shareInformer:        shareInformerFactory.Sharedresource().V1alpha1().Shares().Informer(),
 		listers:              client.GetListers(),
 	}
 
-	client.SetConfigMapsLister(c.informerFactory.Core().V1().ConfigMaps().Lister())
-	client.SetSecretsLister(c.informerFactory.Core().V1().Secrets().Lister())
+	if !readOnly {
+		c.cfgMapWorkqueue = workqueue.NewNamedRateLimitingQueue(
+			workqueue.DefaultControllerRateLimiter(), "shared-resource-configmap-changes")
+		c.secretWorkqueue = workqueue.NewNamedRateLimitingQueue(
+			workqueue.DefaultControllerRateLimiter(), "shared-resource-secret-changes")
+		c.cfgMapInformer = informerFactory.Core().V1().ConfigMaps().Informer()
```

Comment:
if we do employ an ocp csi driver namespace for our config, we would want to always watch that one specifically, even in read only mode

Reply:
I would like to suggest we tackle this change in a standalone pull-request, and also discuss the details in an grooming session.

Reply:
Agreed.  I've added a bullet to the acceptance criteria of the https://issues.redhat.com/browse/BUILD-340 item that @adambkaplan opened for us to capture this

feel free to resolve / consider this done as you see fit @otaviof - thanks